delira/data_loading/data_manager.py

```diff
@@ -762,6 +416,29 @@ def n_batches(self):
 
         truncated_batch = self.n_samples % self.batch_size
 
-        n_batches += int(bool(truncated_batch))
+        n_batches += int(bool(truncated_batch) and not self.drop_last)
 
         return n_batches
+
+    @property
+    def dataset(self):
+        return self.data
+
+    @dataset.setter
+    def dataset(self, new_dset):
+        assert isinstance(new_dset, AbstractDataset)
```

Comment:
Nit: I prefer `TypeErrors` instead of assertion errors, because this actually checks input arguments from the user.
(applies to all checks in this file, `__init__` raises `TypeError` and assertions which is inconsistent)
Furthermore, I think other libraries raise `TypeError` for wrong arguments.